File: apps/api/AssetManagement.Api/Models/ApplicationHistoryChange.cs
namespace AssetManagement.Api.Models;

public class ApplicationHistoryChange
{
    public Guid Id { get; set; }

    public Guid ApplicationHistoryId { get; set; }
    public ApplicationHistory ApplicationHistory { get; set; } = null!;

    public string FieldName { get; set; } = string.Empty;
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
}

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/model/ApplicationHistoryChange.kt
package com.assetmanagement.api.model

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "application_history_changes")
class ApplicationHistoryChange(
    @Id
    @Column(name = "id", columnDefinition = "CHAR(36)")
    var id: UUID = UUID.randomUUID(),

    @Column(name = "application_history_id", nullable = false, columnDefinition = "CHAR(36)")
    var applicationHistoryId: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "application_history_id", insertable = false, updatable = false)
    var applicationHistory: ApplicationHistory? = null,

    @Column(name = "field_name", nullable = false)
    var fieldName: String = "",

    @Column(name = "old_value", columnDefinition = "TEXT")
    var oldValue: String? = null,

    @Column(name = "new_value", columnDefinition = "TEXT")
    var newValue: String? = null
)
